src/commissaire_service/storage/__init__.py

```diff
@@ -0,0 +1,267 @@
+    file will raise an IOError.  Otherwise a default path is tried, but
+    no IOError is raised on failure.  If the file can be opened but not
+    parsed, an exception is always raised.
+
+    :param path: Full path to the config file, or None
+    :type path: str or None
+    :returns: configuration content as a dictionary
+    :rtype: dict
+    :raises: IOError, TypeError, ValueError
+    """
+    json_object = {}
+    using_default = False
+
+    if path is None:
+        path = '/etc/commissaire/commissaire.conf'
+        using_default = True
+
+    try:
+        with open(path, 'r') as fp:
+            json_object = json.load(fp)
+        if using_default:
+            print('Using configuration in {0}'.format(path))
+    except IOError:
+        if not using_default:
+            raise
+
+    if type(json_object) is not dict:
+        raise TypeError('{0}: '.format(path) +
+                        'File content must be a JSON object')
+
+    # Special case:
+    #
+    # In the configuration file, the "authentication-plugin" member
+    # can also be specified as a JSON object.  The object must have
+    # at least a 'name' member specifying the plugin module name.
+    auth_key = 'authentication-plugin'
+    auth_plugin = json_object.get(auth_key)
+    if type(auth_plugin) is dict:
+        if 'name' not in auth_plugin:
+            raise ValueError(
+                '{0}: "{1}" is missing a "name" member'.format(
+                    path, auth_key))
+
+    # Special case:
+    #
+    # In the configuration file, the "storage-handlers" member can
+    # be specified as a JSON object or a list of JSON objects.
+    handler_key = 'storage-handlers'
+    handler_list = json_object.get(handler_key)
+    if type(handler_list) is dict:
+        json_object[handler_key] = [handler_list]
+
+    return json_object
+
+
+class StorageService(CommissaireService):
+    """
+    Provides access to data stores to other services.
+    """
+
+    def __init__(self, exchange_name, connection_url, config_file=None):
+        """
+        Creates a new StorageService and sets up StoreHandler instances
+        according to the config_file.  If config_file is omitted, try
+        the default location (/etc/commissaire/commissaire.conf).
+
+        :param exchange_name: Name of the topic exchange
+        :type exchange_name: str
+        :param connection_url: Kombu connection URL
+        :type connection_url: str
+        :param config_file: Optional configuration file path
+        :type config_file: str or None
+        """
+        queue_kwargs = [ {'routing_key': 'storage.*'} ]
+        CommissaireService.__init__(
+            self, exchange_name, connection_url, queue_kwargs)
+
+        self._manager = StoreHandlerManager()
+
+        # Import the models module so we can look up type names.
+        module_name = 'commissaire.models'
+        module = importlib.import_module(module_name)
```

Comment:
Since this is a static import would it make more sense to do a traditional ``from commissaire import models`` and then use ``models``?